### PR TITLE
Only display rebar in config sidebar when one of rebar, rebar3, or ma…

### LINKED ISSUE #217: Only display supported build tools in Config sidebar
It would be best if `rebar3` was not part of the Config sidebar version list when it is not a package with build tools that include rebar, rebar3 or make. rebar3 will currently filter out any other packages from the registry, which means a user who thinks from the website that they can use a package will find when trying to do so rebar3 complains it does not exist.

### PR DESCRIPTION
**Pull request by bjfish**
…ke is a build tool

Fixes #217

### CODE REVIEW

File: web/templates/package/show.html.eex [hunk 1]
```diff
@@ -6,7 +6,12 @@ licenses = @package.meta["licenses"] || []
 links = Enum.to_list(@package.meta["links"] || [])
 build_tools = @current_release && @current_release.meta["build_tools"] || []
 
-tools = [mix: "mix.exs", rebar: "rebar.config", erlang_mk: "erlang.mk"]
+tools = if Enum.any?(build_tools, fn(tool) -> Enum.member?(["rebar", "rebar3", "make"], tool) end) do
+           [mix: "mix.exs", rebar: "rebar.config", erlang_mk: "erlang.mk"]
+        else
+           [mix: "mix.exs", erlang_mk: "erlang.mk"]
+        end
+
```
Comment (ericmj) [inline web/templates/package/show.html.eex]: Rebar can't build make projects and we shouldn't list erlang.mk when it's a mix project. Can you also move the condition to its own line, it's a bit hard to read as it is now.

Comment (bjfish) [inline web/templates/package/show.html.eex]: @ericmj I've made an update to try to make this more readable and removed make from the rebar compatible function.

Comment (ericmj) [inline web/templates/package/show.html.eex]: It still looks like erlang.mk is displayed for mix projects. I think the code below makes the full compatibility table a bit clearer, what do you think?

```elixir
tools = [mix: "mix.exs", rebar: "rebar.config", erlang_mk: "erlang.mk"]
tool_compatibility = [
  {"mix", [:mix]},
  {"rebar", [:mix, :rebar, :erlang_mk]},
  {"rebar3", [:mix, :rebar, :erlang_mk]},
  {"erlang.mk", [:mix, :erlang_mk]}
]

compatible_tools = Enum.flat_map(build_tools, &(tool_compatibility[&1] || []))
                   |> Enum.uniq

tools = Keyword.take(tools, compatible_tools)
```

We also need to make sure we don't show the config box if no tools are compatible.

Comment (bjfish) [inline web/templates/package/show.html.eex]: I got an error `the Access calls for keywords expect the key to be an atom, got: "mix"`. How about using a map?

```
tools = [mix: "mix.exs", rebar: "rebar.config", erlang_mk: "erlang.mk"]
tool_compatibility = %{
  "mix" => [:mix],
  "rebar" => [:mix, :rebar, :erlang_mk],
  "rebar3"=> [:mix, :rebar, :erlang_mk],
  "erlang.mk" => [:mix, :erlang_mk]
}

compatible_tools = Enum.flat_map( build_tools, &(Map.get(tool_compatibility, &1, [])))
                   |> Enum.uniq

tools = Keyword.take(tools, compatible_tools)
```

I've updated the commit in the PR.

**Comment by ericmj** [comment]
Awesome, thanks!

**Pull request merged**